### ISSUE TITLE
Labels no longer correctly colormapped when added from spinbox

### ISSUE THREAD

**Issue by jni** [napari/napari#2625]
## 🐛 Bug

Unfortunately, another bug snuck in with #2608. Clearly the Labels Qt controls are not well tested in our test suite!

`python examples/add_labels.py` works correctly, showing that this only happens for newly-added labels/colors. But, since #2608, adding labels does not increment the color table:

https://user-images.githubusercontent.com/492549/116353984-198add80-a83b-11eb-90f1-01fc5375f213.mov

This should be fixed before 0.4.8 is released!

```python
import napari
import numpy as np

labels = np.zeros((512, 512), dtype=np.int32)
napari.view_labels(labels)
napari.run()
```

Then, try to increment the label spinbox on the top left corner.

**Comment by DragaDoncila** [comment]
ok so for some reason, this ONLY doesn't work when clicking the buttons or setting the value via the spinbox, but it works when setting the layer `selected_value` programmatically.

If I change `QtLargeIntSpinBox.setValue` to this:

```python
    def setValue(self, value):
        val = self._cast(value)
        super().setValue(val)
        self.valueChanged.emit(val)
 ```
 
then setting the layer controls' `selectionSpinBox.value` programmatically also works fine.

However, clicking the +/- buttons or interacting with the text box doesn't call `QtLargeIntSpinBox.setValue`. However, the docs clearly state that the `valueChanged` signal is emitted whenever the value changes. I think since we've defined our own `valueChanged` signal, the original one now no longer gets emitted.

How do I make sure these buttons and the text box emit our new `valueChanged` signal? @jni @tlambert03 any ideas?

**Comment by tlambert03** [comment]
ahh, ok.  will have a look

**Comment by DragaDoncila** [comment]
nevermind, think I got it lol - instructions [here](https://doc.qt.io/Qt-5/qspinbox.html#singleStep-prop)

"Clicking the up/down buttons or using the keyboard accelerator's up and down arrows will increase or decrease the current value in steps of size singleStep(). If you want to change this behaviour you can reimplement the virtual function stepBy()." 

PR incoming.

**Comment by sofroniewn** [comment]
This was closed by #2648
